### DISCUSSION TITLE
Is the latest version compatible with an old GHES?

### DISCUSSION THREAD

**Discussion by pastelmind** [actions/runner#3445] [Q&A]
I'm working with GHES 3.10.12 (don't ask why). When I attempt to create a self-hosted runner inside GHES, it offers v2.304.0, which was released about a year ago. Would it be OK to use the latest version of the runner (v2.319.1) with our GHES?

**Comment by VigneshAKunder** [comment]
It is not recommended to use the latest runner version (v2.319.1) with your GitHub Enterprise Server (GHES) 3.10.12 instance. 
The version of the self-hosted runner offered by your GHES instance (v2.304.0) is the recommended and compatible version for that specific GHES release. GitHub Enterprise Server versions are designed to work with a specific range of runner versions. 
Trying to use a much newer runner version can lead to compatibility issues, as newer runners may rely on features, APIs, or protocols that are not present in or supported by an older GHES version. For example, a user in a [GitHub discussion](https://github.com/actions/runner/discussions/3136) found that GHES 3.10 only updated the runner to v2.304.0, and newer features (like Node 20 support) were only available after upgrading to GHES 3.12, which then offered runner v2.312.0. 
To ensure stability and support, you should use the runner version provided by your GHES instance or one that is explicitly stated as compatible in the official documentation for GHES 3.10.
If you need the features of a newer runner, you would likely need to upgrade your GHES instance to a newer version first.
